homeassistant/components/shelly/update.py

```diff
@@ -262,6 +261,11 @@ def latest_version(self) -> str | None:
 
         return self.installed_version
 
+    @property
+    def in_progress(self) -> bool | int:
+        """Update installation in progress."""
+        return self._ota_in_progress
```

Comment:
Might be missing something, so what is the advantage over just using `_attr_in_progress`?
It's also `bool | int`, initially `False`, and seems to provide the same functionality as the Shelly-specific `_ota_in_progress` attribute.

Reply:
If we use `_attr_in_progress`, the progress visualization will stop here https://github.com/home-assistant/core/blob/c2ffed9b2defdc9861b9f0e51987ae7aadd12659/homeassistant/components/update/__init__.py#L469
Our `async_install()` method doesn't actually install the update but only trigger the device to start the update and we only get install progress by events.

Reply:
Ah, thanks.